dashboard/app/assets/javascripts/dialog.js.erb

```diff
@@ -54,15 +54,41 @@ function Dialog(options) {
         $(this).remove();
       });
 
+  $(this.div).on(
+      'hide.bs.modal',
+      function(e) {
```

Comment:
I think you should be able to bind a function here to the Dialog instance so you don't need to do two things:
(1) store the object reference on the div
(2) store the hideOptions on the div
I think it would be cleaner to store everything in the Dialog object.